src/group_theory/perm/basic.lean

```diff
@@ -267,6 +288,30 @@ instance perm_unique {n : Type*} [unique n] : unique (equiv.perm n) :=
 
 @[simp] lemma default_perm {n : Type*} : default (equiv.perm n) = 1 := rfl
 
+variables (e : perm α) (ι : α ↪ β)
+
+open_locale classical
+
+/-- Noncomputable version of `equiv.perm.via_embedding`, but does not assume `fintype` -/
+noncomputable def of_embedding : perm β :=
+extend_domain e (of_injective ι.1 ι.2)
```

Comment:
I guess probably `via_embedding` should be renamed to `via_fintype_embedding`, and then this can take the `via_embedding` name for itself?